data/src/test/java/com/microsoft/azure/kusto/data/KustoResultMapperTest.java

```diff
@@ -0,0 +1,246 @@
+			.addNullableColumn(KustoType.STRING, "Enum", (r, t) -> r.setTestEnum(t != null ? TestEnum.valueOf(t) : null))
+			.addNullableColumn(KustoType.DATETIME_INSTANT, "Instant", TestPojo::setInstantTime).build();
+	
+	static final KustoResultMapper<TestPojo> ordinalMapper = KustoResultMapper.newBuilder(TestPojo::new)
+			.addNonNullableColumn(KustoType.INTEGER, 1, TestPojo::setId).addNullableColumn(KustoType.STRING, 2, TestPojo::setName)
+			.addNullableColumn(KustoType.DATETIME_ZONED_DATE_TIME, 3, TestPojo::setZonedTime).addNullableColumn(KustoType.GUID_UUID, 4, TestPojo::setUuid)
+			.addNullableColumn(KustoType.STRING, "Enum", (r, t) -> r.setTestEnum(t != null ? TestEnum.valueOf(t) : null))
+			.addNullableColumn(KustoType.DATETIME_INSTANT, 6, TestPojo::setInstantTime).build();
+	
+	static final KustoResultMapper<TestPojo> mixedMapper = KustoResultMapper.newBuilder(TestPojo::new)
+			.addNonNullableColumn(KustoType.INTEGER, "Id", 1, TestPojo::setId).addNullableColumn(KustoType.STRING, "Name", 2, TestPojo::setName)
+			.addNullableColumn(KustoType.DATETIME_ZONED_DATE_TIME, "TimeColumn", 3, TestPojo::setZonedTime)
+			.addNullableColumn(KustoType.GUID_UUID, "Guid", 4, TestPojo::setUuid)
+			.addNullableColumn(KustoType.STRING, "Enum", 5, (r, t) -> r.setTestEnum(t != null ? TestEnum.valueOf(t) : null))
+			.addNullableColumn(KustoType.DATETIME_INSTANT, "Instant", 6, TestPojo::setInstantTime).build();
+	
+	KustoResultSetTable resultSet = mock(KustoResultSetTable.class);
+	
+	@BeforeEach
+	public void prepareResultSet() {
+		when(this.resultSet.findColumn("Id")).thenReturn(1);
+		when(this.resultSet.findColumn("Name")).thenReturn(2);
+		when(this.resultSet.findColumn("TimeColumn")).thenReturn(3);
+		when(this.resultSet.findColumn("Guid")).thenReturn(4);
+		when(this.resultSet.findColumn("Enum")).thenReturn(5);
+		when(this.resultSet.findColumn("Instant")).thenReturn(6);
+	}
+	
+	void testNonNull(KustoResultMapper<TestPojo> mapper) throws Exception {
+		when(this.resultSet.getObject(1)).thenReturn(Integer.valueOf(1));
+		when(this.resultSet.getObject(2)).thenReturn("MyName");
+		when(this.resultSet.getObject(3)).thenReturn("1970-01-01T00:00:00.001Z");
+		when(this.resultSet.getObject(4)).thenReturn("e091cf92-6195-4005-bad5-82af80ff1939");
+		when(this.resultSet.getObject(5)).thenReturn("C");
+		when(this.resultSet.getObject(6)).thenReturn("1970-01-01T00:00:00.002Z");
+		when(this.resultSet.next()).thenReturn(true);
+		
+		TestPojo pojo = mapper.extractSingle(this.resultSet);
+		assertNotNull(pojo);
+		assertEquals(1, pojo.id);
+		assertEquals("MyName", pojo.name);
+		assertEquals(1, pojo.zonedTime.toInstant().toEpochMilli());
+		assertEquals(UUID.fromString("e091cf92-6195-4005-bad5-82af80ff1939"), pojo.uuid);
+		assertEquals(TestEnum.C, pojo.testEnum);
+		assertEquals(2, pojo.instantTime.toEpochMilli());
+	}
+	
+	void testNullable(KustoResultMapper<TestPojo> mapper) throws Exception {
```

Comment:
testSingleNullable

Reply:
renamed all tests that test "extractSingle" - added "single" to all of them.